packages/components/src/List/ListComposition/SortBy/SortBy.component.js

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Navbar, NavDropdown, Nav, NavItem, MenuItem } from 'react-bootstrap';
+import uuid from 'uuid';
+import isUndefined from 'lodash/isUndefined';
+
+import { useListContext } from '../context';
+
+/**
+ * Get the component's selected sort by option
+ * @param {object} props Props passed to the component
+ * @param {object} sortParams Current value of the context's `sortParams`
+ * @returns {string}
+ */
+function getSelectedOption(props, sortParams) {
+	if (props.selected) {
+		return props.selected; // From props (controlled mode)
+	} else if (sortParams.sortBy) {
+		return sortParams.sortBy; // From context (uncontrolled)
+	}
+	return props.options[0].key; // Default to first available option
+}
+
+function SortBy(props) {
+	const { id, options, selected, isDescending: isDescendingProp, onChange, onOrderChange } = props;
+	const { sortParams, setSortParams, t } = useListContext();
+
+	const selectedOption = getSelectedOption(props, sortParams);
+
+	useEffect(() => {
+		if (!selected && sortParams.sortBy) {
+			// Set context's sortBy parameter if there's no value provided (prop or context)
+			setSortParams({ ...sortParams, sortBy: selectedOption });
+		}
+	});
+
+	const onSelect = onChange
+		? (value, event) => onChange(event, value)
+		: value => setSortParams({ ...sortParams, sortBy: value });
+
+	const selectedLabel = options.find(option => option.key === selectedOption).label;
+
+	// Sort order
+	const isDescending = !isUndefined(isDescendingProp) ? isDescendingProp : sortParams.isDescending;
```

Comment:
Suggestion: instead of checking the presence of different props at different position in the code, you can create a boolean `const controlledMode = props.onChange && props.onOrderChange`.

In general, if there are the callbacks, we can consider that we are in controlled mode.
Let's not allow a mix of controlled for a value and uncontrolled for another one.